Vagrantfile

```diff
@@ -117,6 +130,16 @@ echo "Launching bootstrap"
 exec "/srv/scality/metalk8s-$SHORT_VERSION/bootstrap.sh"
 SCRIPT
 
+DEPLOY_SSH_PUBLIC_KEY = <<-SCRIPT
+#!/bin/bash
+
+set -eu -o pipefail
+
+if ! grep -Fxq "$(cat .ssh/#{PRESHARED_SSH_KEY_NAME}.pub)" .ssh/authorized_keys ; then
+   cat .ssh/#{PRESHARED_SSH_KEY_NAME}.pub >> .ssh/authorized_keys
```

Comment:
Instead of copying the file into the VM, could as well read it in Ruby, then

```sh
cat >> .ssh/authorized_keys << EOF
#{SSH_PUBLIC_KEY}
EOF
```

(and do the same for the private key).

Reply:
+1

Can be read with `File.read(path, :encoding => 'utf-8'))` or, if we are using Ruby >= 1.9: `File.read(path, encoding: 'utf-8')`

Reply:
@nootal I think the benefit of that approach is that the `Vagrantfile` script could error out, before starting to deploy something, which could be useful if someone runs `vagrant up` (not `doit vagrantup`), and the key hasn't been created before.